test/model/tpp/test_tpp_distribution_inference.py

```diff
@@ -0,0 +1,187 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""
+Test that maximizing likelihood allows to correctly recover distribution parameters for all
+distributions exposed to the user.
+"""
+# Standard library imports
+from typing import List
+
+# Third-party imports
+import mxnet as mx
+import numpy as np
+import pytest
+from pydantic import PositiveFloat, PositiveInt
+
+# First-party imports
+from gluonts.model.common import NPArrayLike
+from gluonts.model.tpp.distribution import (
+    Loglogistic,
+    LoglogisticOutput,
+    Weibull,
+    WeibullOutput,
+    TPPDistributionOutput,
+)
+
+NUM_SAMPLES = 2000
+BATCH_SIZE = 32
+TOL = 0.3
+START_TOL_MULTIPLE = 1
+
+np.random.seed(1)
+mx.random.seed(1)
+
+
+def inv_softplus(y: NPArrayLike) -> np.ndarray:
+    # y = log(1 + exp(x))  ==>  x = log(exp(y) - 1)
+    return np.log(np.exp(y) - 1)
+
+
+def maximum_likelihood_estimate_sgd(
```

Comment:
does this have any major difference from its original? if not can we get away with importing / slightly changing the original?